plugins/tangle/solidifier.go

```diff
@@ -90,17 +96,26 @@ func registerApproverOfApprovee(approver trinary.Hash, approveeHash trinary.Hash
 // Can be aborted with abortSignal
 func solidQueueCheck(milestoneIndex milestone_index.MilestoneIndex, cachedMsTailTx *tangle.CachedTransaction, abortSignal chan struct{}) (solid bool, aborted bool) {
 
-	defer cachedMsTailTx.Release() // tx -1
-
 	ts := time.Now()
 
-	txsChecked := make(map[string]bool) // isSolid
-	txsToTraverse := make(map[string]struct{})
-	approvers := make(map[string]map[string]struct{})
-	entryTxs := make(map[string]struct{})
-	txsToRequest := make(map[string]struct{})
+	cachedTxs := make(map[trinary.Hash]*tangle.CachedTransaction)
+	cachedTxs[cachedMsTailTx.GetTransaction().GetHash()] = cachedMsTailTx
+
+	defer func(cachedTxs map[trinary.Hash]*tangle.CachedTransaction) {
+		// Release all txs at the end
+		for _, cachedTx := range cachedTxs {
+			cachedTx.Release() // tx -1
+		}
+	}(cachedTxs)
```

Comment:
not sure if cachedTxs is captured here or at the end, I think it is captured at this line